src/main/java/net/glowstone/block/entity/FlowerPotEntity.java

```diff
@@ -25,15 +25,14 @@ public FlowerPotEntity(GlowBlock block) {
     @Override
     public void loadNbt(CompoundTag tag) {
         super.loadNbt(tag);
-
-        int contentsData = tag.isInt("Data") ? tag.getInt("Data") : 0;
-
-        if (tag.isString("Item") && !tag.getString("Item").isEmpty()) {
-            // NBT data uses material ID names (post-1.8).
-            contents = ItemIds.getItem(tag.getString("Item")).getNewData((byte) contentsData);
-        } else if (tag.isInt("Item")) {
+        final Material[] item = {null};
+        // NBT data uses material ID names (post-1.8).
+        if (!tag.readString("Item", itemString -> item[0] = ItemIds.getItem(itemString))) {
             // NBT data uses material IDs (pre-1.8).
-            contents = Material.getMaterial(tag.getInt("Item")).getNewData((byte) contentsData);
+            tag.readInt("Item", itemInt -> item[0] = Material.getMaterial(itemInt));
+        }
+        if (item[0] != null) {
```

Comment:
I'm not sure if I like this. Could `readX` allow for chaining with `andThen` or similar (maybe we need a method for failures as well). Then if it's still necessary, we could move the bool that returns if the tag exists to a new method.